impeller/tessellator/tessellator_unittests.cc

```diff
@@ -484,5 +484,18 @@ TEST(TessellatorTest, FilledRoundRectTessellationVertices) {
        Rect::MakeXYWH(5000, 10000, 2000, 3000), {50, 70});
 }
 
+#if !NDEBUG
+TEST(TessellatorTest, ChecksConcurrentPolylineUsage) {
+  auto tessellator = std::make_shared<Tessellator>();
+  PathBuilder builder;
+  builder.AddLine({0, 0}, {100, 100});
+  auto path = builder.TakePath();
+
+  auto polyline = tessellator->CreateTempPolyline(path, 0.1);
+  EXPECT_DEBUG_DEATH(tessellator->CreateTempPolyline(path, 0.1),
+                     "point_buffer_");
```

Comment:
Is this something that could happen in release builds?  Should this return a `StatusOr<Path::Polyline>`?

Reply:
It could happen in the same way that an NPE or off by one in an array could happen. I'm not sure. if this is happening its because we have a severe bug. Given that the tessellator is already documented as being thread unsafe I'd prefer to leave it as is.